### ISSUE TITLE
Option to give statics based on topic

### ISSUE THREAD

**Issue by soloam** [bartbutenaers/node-red-contrib-msg-speed#13]
Hello, nice work on this node!

Would it be possible to give statics based on topic?

Tks

**Comment by bartbutenaers** [comment]
Hi @soloam,
I have recently completely rewritten this node, which will make it easier to implement things like ***topic dependent*** speed measurements.

However - due to a surgery - I'm not able to develop for a couple of months.
I will get back to you afterwards!
Bart

**Comment by soloam** [comment]
Thank you @bartbutenaers! Hope you get to feeling better soon!

**Comment by bartbutenaers** [comment]
Morning @soloam,

I'm back ;-)
This repository now has a ***"topic-dependent"*** branch, which offers functionality to control the statistics per topic.

It would be nice if you could do some tests, and read through the [readme](https://github.com/bartbutenaers/node-red-contrib-msg-speed/tree/topic-dependent) page to see if the documentation is clear.  Of course the others that have liked your request, are also welcome ...
That would be a ***great*** help, because I have a large backlog due to my absence...

You can install that branch, by executing this command within your .node-red folder:
```
npm install bartbutenaers/node-red-contrib-msg-speed#topic-dependent
```

I'm not very pleased yet with the ***node status***, so please don't hesitate to share thoughts!  Suppose you have 10 topics and you send e.g. a pause message for 1 of those topics, then the node status text becomes ***"paused"***.  In the old version that was correct because the node was paused for all messages.  But now it is only paused for 1 message, but not paused for the other 9 topics.  So it doesn't really makes sense anymore ...  Perhaps I should ask this on the Node-RED Discourse forum, to collect ideas...

P.S. The new readme page contains a flow that you can use and expand to test this functionality.  All the example flows are now also available via the Node-RED flow editor (Import menu -> examples).  And there is a link to that menu in the node documentation (Info panel in the flow editor).

Kind regards,
Bart

**Comment by bartbutenaers** [comment]
Reminder for myself: once this feature works correctly, add the same behaviour to the node-red-contrib-msg-size node!

**Comment by bartbutenaers** [comment]
@soloam,

I have refactored the code a bit more, and now the node status seems better to me:

![msg_speed_status](https://user-images.githubusercontent.com/14224149/122656879-e7249f00-d15e-11eb-9a94-54162a99ffbe.gif)

It would be nice if you could test it, to make sure I haven't forgotten anything.  Because the code has been changed quite heavily, and this node is being used a lot.

Bart

**Comment by soloam** [comment]
Hello... I have this on my to-do list, but lacking in time! I'll try to do it this week! Thank you for your time on this

**Comment by soloam** [comment]
Hello @bartbutenaers I've been testing your node for some days and it seams to be working ok! You did a great job on this one! Tks

**Comment by bartbutenaers** [comment]
Hi @soloam,

Thank you for testing and getting back here to give feedback!!
It is now available as version ***2.1.0*** in the palette:

![image](https://user-images.githubusercontent.com/14224149/124672284-10f7f880-deb7-11eb-81f7-37ea18603aaa.png)
